lib/Utils.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const Path = require('path');
+const Config = require('./config');
+
+const Utils = {
+    async createConfigResolver(options) {
+        // Some file watchers only watch a file if it's been require'd.
+        // Determination doesn't call "require()" on the file internally,
+        // so by "require"-ing here, we allow for file watching.
+        // Parsing is expected to fail since manifests allow for comments.
+        Utils.safeRequire(options.config);
+
+        // Create and return the resolver
+        return await Config.resolve(options);
+    },
+
+    safeRequire(name) { // eslint-disable-line consistent-return
+
+        try {
+            return require(name);
+        }
+        catch (e) {
+            // noop
+        }
+    },
+
+    async mergeManifestConfigs(options) {
+
+        const {
+            basedir,
+            config,
+            environment,
+            protocols,
+            userConfigPaths
+        } = options;
+
+        const baseConfigBaseDir = Path.dirname(config);
+
+        const resolved = await Config.resolve({
+            basedir: baseConfigBaseDir,
```

Comment:
The previous code allowed `basedir` to be specified by the options. It looks like here we're always using `Path.dirname(config)` rather than the `basedir` passed into the options.

Reply:
I'm doing that here to safely set up the base config that the userConfigPaths will subsequently be merged into. This i because if an arbitrary base dir is passed in and applied to the base config in catayst-server then it will error out because it can't find the plugins in the base config using their relative paths since they are now expected to be in the basedir and not where they actually are which is in that library's directory. 

Hoping to find a way around this if I can somehow  implement the suggestions above re: consolidating userConfigPaths into config and passing to steerage.